### ISSUE TITLE
Add SpiderClusterOptions to x-map-marker-layer component

### ISSUE THREAD

**Issue by BertCanavaggio** [infusion-code/angular-maps#71]
Hi,
Can you easily add the SpiderClusterOptions input to the x-map-marker-layer component?
Thanks!

**Comment by thor-schueler** [comment]
It should not be that hard to subclass and implement. However, I am not sure it makes a lot of sense. We added the x-map-marker-layer specifically to accommodate large sets of markers relatively performant (as in 10s of thousands). With that number of markers, spidering clusters don't make too much sense as each cluster will have potentially many hundreds or even thousands of markers, so the spider would not be very useful. Spidering works well with small numbers of markers in a cluster (think <100) from a usability perspective.

If you do not need to have the scale that the x-map-layer component provides, you can simple use the x-cluster-layer component. 

Perhaps you could provide a bit more insight into what you would like to achieve, and I can perhaps provide some more guidance.

**Comment by BertCanavaggio** [comment]
Thanks for your quick answer. 

I have 1000 markers and I would need to spider some small clusters after zooming in. So far, I've used x-map-marker-layer which seems to run much faster than x-cluster-layer (I can give it another try thought). 

I noticed x-map-marker-layer supports ClusterClickAction.Spider as an Input, so I was wondering if we could bind SpiderClusterOptions as well.

**Comment by thor-schueler** [comment]
Yep, marker layer is much faster indeed. But to achieve the speed, we drop a lot of overhead. 

The only reason why ClusterClickAction.Spider is available though is that it is part of the model. We are only supporting ZoomIntoCluster at the moment.  There is a bunch of logic that is implemented in https://github.com/infusion-code/angular-maps/blob/master/src/models/bing/bing-cluster-layer.ts to support spiders which is currently not available in the x-map-marker-layer. I could certainly be done, but would require a bit of work. Don't think it would be too hard though. You could take a crack at it and create a pull request. Essentially, it would involve changes to  https://github.com/infusion-code/angular-maps/blob/master/src/components/map-marker-layer.ts lines 369-380 to attache the spider cluster options. It might actually pretty straight forward, now that I think about it ;)
